tools/libclang/CRefactor.cpp

```diff
@@ -1884,13 +1884,18 @@ clang_IndexerQuery_consumeIntResult(CXIndexerQuery Query, unsigned CursorIndex,
     return CXIndexerQueryAction_None;
   if (Wrapper->DeclResults.empty())
     Wrapper->DeclResults.resize(DQ->getInputs().size(),
-                                PersistentDeclRef<Decl>::create(nullptr));
+                                indexer::Indexed<PersistentDeclRef<Decl>>(
+                                    PersistentDeclRef<Decl>::create(nullptr),
+                                    indexer::QueryBoolResult::Unknown));
   // Filter the declarations!
   bool IsNot = false;
   if (isa<indexer::detail::DeclPredicateNotPredicate>(DQ->getPredicateNode()))
     IsNot = true;
-  Wrapper->DeclResults[CursorIndex] = PersistentDeclRef<Decl>::create(
-      (IsNot ? !Value : !!Value) ? DQ->getInputs()[CursorIndex] : nullptr);
+  bool Result = IsNot ? !Value : !!Value;
+  Wrapper->DeclResults[CursorIndex] = indexer::Indexed<PersistentDeclRef<Decl>>(
+      PersistentDeclRef<Decl>::create(Result ? DQ->getInputs()[CursorIndex]
+                                             : nullptr),
+      Result ? indexer::QueryBoolResult::Yes : indexer::QueryBoolResult::No);
```

Comment:
Just to double check: if you construct an Indexed(DQ->getInputs()[CursorIndex], QueryBoolResult::Yes), are you saying that the declaration is not defined in the entire project?